tools/dds/dds-adapter/lrs-device-controller.cpp

```diff
@@ -230,44 +231,55 @@ std::vector< std::shared_ptr< realdds::dds_stream_server > > lrs_device_controll
         server->init_profiles( profiles, default_profile_index );
 
         // Get supported options and recommended filters for this stream
-        realdds::dds_options options;
+        realdds::dds_options stream_options;
         std::vector< std::string > filter_names;
         for( auto sensor : _rs_dev.query_sensors() )
         {
             std::string const sensor_name = sensor.get_info( RS2_CAMERA_INFO_NAME );
-            if( server->sensor_name().compare( sensor_name ) == 0 )
+            if( server->sensor_name().compare( sensor_name ) != 0 )
+                continue;
+
+            // Multiple streams map to the same sensor so we may go over the same sensor multiple times - we
+            // only need to do this once per sensor!
+            if( sensors_handled.emplace( sensor_name ).second )
```

Comment:
So we will have all the options in one stream, e.g. depth, and none in others of the same sensor, e.g Infrared 1/2 ?
We want the streams independant. Any way, realdds should not use the sensor concept.
One stream changing option value for all the sensor is a current RealSense cameras limitation that
  1. Should be addressed at librealsense level (like here in the adapter)
  2. May not always be true for all camera models

Reply:
I don't understand the objection.
Yes, we're sensor-based, and therefore having per-sensor options is going to be true for all cameras.

That means the adapter can make the assumption that sensor options are for all streams, and therefore it has to choose one to put the options in. Repeating them for all streams does not make sense. Options should not be duplicated, they should be specific. It further makes sense that the depth sensor options would go in the depth stream (though notice I did not do that).

It is addressed at the librealsense level - we assign the options to the sensor proxy, regardless of which stream they came from. That's the model today, and we may change it in the future. But, for the purposes of adapting a USB camera that's sensor-based to DDS, only one set of options is needed.

The eth cam can implement options any way we want, including IR-specific options that do not exist for depth. Librealsense will still stick all of them into the same sensor. But at least now, when you set an option, it'll set one and only one, and in the right stream.

Reply:
But what if you are only using Depth stream but all the options are registered through the Infrared stream? You will not see any options to set for your stream.

Lets discuss this tomorrow.